src/main/drivers/compass/compass_ak8975.c

```diff
@@ -135,8 +135,9 @@ static bool ak8975Read(magDev_t *mag, int16_t *magData)
     switch (state) {
         default:
         case STATE_READ_STATUS1:
-            busReadRegisterBufferStart(dev, AK8975_MAG_REG_ST1, &status, sizeof(status));
-            state = STATE_WAIT_STATUS1;
+            if (busReadRegisterBufferStart(dev, AK8975_MAG_REG_ST1, &status, sizeof(status))) {
+                state = STATE_WAIT_STATUS1;
+            }
```

Comment:
Maybe code it this way (continue only is operation succeeds):

```suggestion
            if (!busReadRegisterBufferStart(dev, AK8975_MAG_REG_ST1, &status, sizeof(status))) {
                return false;
            }
            state = STATE_WAIT_STATUS1;
```

Reply:
That would still require a `return false` following the state assignment.

Reply:
The logic is:
If operation fails, return false to retry soon
Set next state
Return false to call next state faster

first and second false have a bit different meaning.
And it is more similar to longer cases - return of something fails, continue othervise.

But resulting instructions are the same and your variant is shorter. It is only minor stylistics suggestion..